MITADS/opensubtitles_exporter.py

```diff
@@ -0,0 +1,367 @@
+
+          ## Generate multiple files if split_output_file_rows is enabled
+          if split_output_file_rows!=None and not (count_sentence % split_output_file_rows):
+            #count_sentence = 0
+            file_output.close()
+            new_filename = "opensubtiles_extracted_{}.txt".format(str(count_split_file))
+            file_output=open(new_filename,"w")
+            print('NEW FILE OUTPUT: ' + new_filename)
+            count_split_file +=1
+          ####
+          try:
+            ##decomment if you want to check input vs output
+            #file_output.write(unidecode(_line))         
+               
+            file_output.write(line)
+            file_output.write('\n')
+            count_sentence +=1
+          except Exception as e:
+            raise(e)
+          
+          exported_file = True
+
+        ##If it is important to separate sentences based on waiting breaks, 
+        # decomment this... line with time tag start with '<time'
+        #if(line.startswith('<time')):          
+        #  prev_line = line
+      
+      if(exported_file):
+        ##file exported line separator 
+        file_output.write('\n')
+    finally:
+      fp.close()
+      pass
+
+    count_file +=1
+
+    ##progress print
+    if not (count_file % 100):
+      print (count_file, '/', total_files, 'file processed')
+
+  if(file_output!=None):
+      file_output.close()
+
+  print ('\n')
+  print (count_sentence, ' sentences extracted.' )
+  
+def process_text(text):
+  ##clear and normalize text
+  text = text.strip()
+
+  ##exclude line not relevant
+  if(line_not_relevant(text)):
+    return ''
+
+  ##normalization
+  text = normalize_text(text)
+
+  return text
+
+def test_regexp():
+  
+  #test cases found in opensubtitles dataset.
+
+  #this function could be used for a test case of text processing (clear and normalization),
+  #to be implemented better...
+
+  test_pair = []
+  test_pair.append(['Per cominciare... la civilta\' moderna e\' una truffa.','Per cominciare. la civiltà moderna è una truffa.'])
+  test_pair.append(['£ 500 in premio per i ladri di cavalli','500 lire in premio per i ladri di cavalli'])
+  test_pair.append(['{\pos(346372)}Agisco senza emozione','Agisco senza emozione'])
+  test_pair.append(["Si', e' cosi'.",'Sì, è così.'])
+  test_pair.append(["Si'... ",'Sì.'])
+  test_pair.append(["testo ddjd(( -- )) - [ ok... ",'testo ddjd ok.'])
+  test_pair.append(["Com'e?",'Com\'è?'])
+ 
+
+  succ = True
+  for pair in test_pair:
+    text = pair[0]
+    text_res =  process_text(text)
+    text_expected = pair[1]
+    if(text_res!=text_expected):
+      print('error on text: {}'.format(text))
+      print('output produced: {}'.format(text_res))
+      print('output expected: {}'.format(text_expected)) 
+      print('\n')
+      succ = False
+  
+  if(succ):
+    print('Processing successfully!!!!')
+
+if __name__ == '__main__':
+
+  ##decomment to run some test case on regexp expression
+  #test_regexp()
```

Comment:
Can we remove this and the function?

Reply:
this function was useful.  I only remove the commented code.